src/datachain/lib/file.py

```diff
@@ -272,8 +272,12 @@ def read_text(self):
     def save(self, destination: str):
         """Writes it's content to destination"""
         destination = stringify_path(destination)
-        client: Client = self._catalog.get_client(str(destination))
-        client.upload(self.read(), str(destination))
+        client: Client = self._catalog.get_client(destination)
+
+        if client.PREFIX == "file://" and not destination.startswith(client.PREFIX):
+            destination = client.PREFIX + os.path.abspath(destination)
```

Comment:
This is still not correct. `file://` URI uses forward-slash as a separator, but by using `os.path`, the separator could be a backward-slash (`\\`) here.

You can use either of the options:
```python
from pathlib import Path
Path(destination).absolute().as_uri()
```

or, `urllib.request.pathname2url` or, `make_path_posix` from fsspec.